mne/viz/evoked.py

```diff
@@ -1819,18 +1912,39 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
             head_pos = {'center': (0, 0), 'scale': (0.5, 0.5)}
             pos, outlines = _check_outlines(pos, np.array([1, 1]), head_pos)
             if not isinstance(show_sensors, (np.int, bool)):
-                raise TypeError("`show_sensors` must be numeric or bool, not" +
-                                str(type(show_sensors)))
+                raise ValueError("show_sensors must be numeric or bool, not" +
+                                 str(type(show_sensors)))
             if show_sensors is True:
                 show_sensors = 2
             _plot_legend(pos, ["k" for pick in picks], axes, list(), outlines,
                          show_sensors, size=20)
 
-    if show_legend and len(conditions) > 1:
+    # the condition legend
+    if len(conditions) > 1:
         if show_legend is True:
             show_legend = 'best'
-        axes.legend(loc=show_legend, ncol=1 + (len(conditions) // 5),
-                    frameon=True)
+        legend_params = dict(loc=show_legend, frameon=True)
+        if split_legend:
+            if len(legend_lines) > 1:
+                axes.legend(legend_lines, legend_labels,  # see above: mpl 1.3
+                            ncol=1 + (len(legend_lines) // 4), **legend_params)
+        else:
+            axes.legend(ncol=1 + (len(conditions) // 5), **legend_params)
+
+    # the colormap, if `cmap` is provided
+    if split_legend and cmap is not None:
+        # plot the colorbar ... complicated cause we don't have a heatmap
+        from mpl_toolkits.axes_grid1 import make_axes_locatable
+        divider = make_axes_locatable(axes)
+        ax_cb = divider.append_axes("right", size="5%", pad=0.05)
+        ax_cb.imshow(the_colors[:, np.newaxis, :], interpolation='none')
+        ax_cb.set_yticks(np.arange(len(the_colors)))
+        ax_cb.set_yticklabels(np.array(color_conds)[color_order])
+        ax_cb.yaxis.tick_right()
+        ax_cb.set_xticks(())
+        ax_cb.set_ylabel(cmap_label)
+        fig.cbar = ax_cb
+    fig.ts_ax = axes
```

Comment:
Oh, that's nice, I wasn't aware you can add custom properties to Figure.

Reply:
Errrr. That's just monkey patching stuff I think. It's not supported at all, I'm just sticking it on there.

Reply:
It's better not to monkey patch unless you have a good reason / really need to. I suspect your colorbar axes can still be accessed via `figure.axes` somewhere so this doesn't seem worth it

Reply:
I mean, it's not exposed. It's just for making it easier internally.

I would prefer to avoid figure.axes cause we have a varying number of axes.

Reply:
> I would prefer to avoid figure.axes cause we have a varying number of axes.

But based on when it's created it should always be the last axis of the figure. It seems cleaner/more standard to get it as `figure.axes[-1]`.